src/util/strencodings.cpp

```diff
@@ -98,20 +98,22 @@ std::optional<std::vector<Byte>> TryParseHex(std::string_view str)
 template std::optional<std::vector<std::byte>> TryParseHex(std::string_view);
 template std::optional<std::vector<uint8_t>> TryParseHex(std::string_view);
 
-bool SplitHostPort(std::string_view in, uint16_t& portOut, std::string& hostOut)
+bool SplitHostPort(std::string_view in, uint16_t& portOut, std::string& hostOut, bool* cPortOut)
```

Comment:
Not a huge fan of this new parameter. The return value has almost exactly what we need, can we reuse somehow?